jetbrains-core/src/software/aws/toolkits/jetbrains/settings/AwsSettingsConfigurable.java

```diff
@@ -80,10 +85,29 @@ public boolean isModified() {
     }
 
     @Override
-    public void apply() {
+    public void apply() throws ConfigurationException {
         SamSettings samSettings = SamSettings.getInstance();
-        LambdaSettings lambdaSettings = LambdaSettings.getInstance(project);
 
+        String path = getSamExecutablePath();
+        // if user path is empty
+        if (path == null || path.isEmpty()) {
+            // try to autodetect the path
+            path = samSettings.getExecutablePath();
+
+            // if path is still empty pop the error
+            if (path == null || path.isEmpty()) {
+                throw new ConfigurationException(message("lambda.run_configuration.sam.not_specified"));
```

Comment:
Can we try throwing `com.intellij.execution.configurations.RuntimeConfigurationError` like we do here https://github.com/aws/aws-toolkit-jetbrains/blob/master/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamRunConfiguration.kt#L152

Reply:
RuntimeConfigurationError extends ConfigurationException and results in the same style as in the demo posted above :(

Reply:
:( I was afraid of that.